### DISCUSSION TITLE
Flutter pag中，有概率出现被看门狗杀死的崩溃，辛苦排查一下

### DISCUSSION THREAD

**Discussion by zhch19910514** [Tencent/libpag#2015] [Q&A]
#0 Thread Crashed
signal: 9
terminationReason: <RBSTerminateContext| domain:10 code:0x8BADF00D explanation:scene-update watchdog transgression: application<com.tencent.qqreaderiphone>:55345 exhausted real (wall clock) time allowance of 10.00 seconds ProcessVisibility: Foreground ProcessState: Running WatchdogEvent: scene-update WatchdogVisibility: Background WatchdogCPUStatistics: ( "Elapsed total CPU time (seconds): 7.630 (user 7.630, system 0.000), 12% CPU", "Elapsed application CPU time (seconds): 0.336, 1% CPU" ) reportType:CrashLog maxTerminationResistance:Interactive>
0 dyld start + 2528
1 QQReaderUI main (main.m:45)
2 QQReaderUI qrapm_UIApplicationMain (QRMatrixToolLaunchMainFuncMonitor.m:32)
3 QQReaderUI pld_UIApplicationMain (PLDAppDelegateGuard.m:26)
4 UIKitCore _UIApplicationMain + 340
5 UIKitCore -[UIApplication _run] + 888
6 GraphicsServices _GSEventRunModal + 164
7 CoreFoundation _CFRunLoopRunSpecific + 612
8 CoreFoundation ___CFRunLoopRun + 2036
9 CoreFoundation ___CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__ + 16
10 libdispatch.dylib __dispatch_main_queue_callback_4CF + 44
11 libdispatch.dylib __dispatch_main_queue_drain + 928
12 libdispatch.dylib __dispatch_client_callout + 20
13 libdispatch.dylib __dispatch_call_block_and_release + 32
14 Flutter ___ZN7flutter25PlatformMessageHandlerIos21HandlePlatformMessageENSt21_LIBCPP_ABI_NAMESPACE10unique_ptrINS_15PlatformMessageENS1_14default_deleteIS3_EEEE_block_invoke (ref_ptr.h:110)
15 Flutter __45-[FlutterMethodChannel setMethodCallHandler:]_block_invoke (FlutterChannels.mm:277)
16 QQReaderUI -[FlutterPagPlugin handleMethodCall:result:] (FlutterPagPlugin.m:0)
17 QQReaderUI -[FlutterPagPlugin initPag:result:] (FlutterPagPlugin.m:153)
18 QQReaderUI -[FlutterPagPlugin pagRenderWithPagData:progress:repeatCount:autoPlay:result:] (FlutterPagPlugin.m:0)
19 QQReaderUI -[TGFlutterPagRender initWithPagData:progress:autoPlay:frameUpdateCallback:] (TGFlutterPagRender.mm:86)
20 libpag 4405223676
21 libpag 4405223892
22 libpag 4405256576
23 libpag 4405499756
24 libpag 4404685920
25 libpag 4405606636
26 libpag 4405326632
27 libpag 4404996452
28 libpag 4405504712
29 libc++.1.dylib std::__1::condition_variable::wait(std::__1::unique_lock<std::__1::mutex>&) + 28
30 libsystem_pthread.dylib __pthread_cond_wait + 1232
31 libsystem_kernel.dylib ___psynch_cvwait + 8

**Comment by gdoujk1123ljm** [comment]
提供一下你们使用的代码，或者独立的复现代码

**Reply by lzou** [reply]
Flutter在播放pag动画的时候，在原生打断点二十秒左右几乎必现，
排查是
CVPixelBufferRef target = [_surface getCVPixelBuffer];

**Comment by CodeJhF** [comment]
感谢 @gdoujk1123ljm 的回复。

因长期无回复，且 libpag 4.3 及更早版本已停止维护（当前仅维护 4.4 / 4.5），先关闭本条。
上报版本：0.000 / 0.336。

如果问题在最新版本（4.4 / 4.5）仍能复现，欢迎在本帖回复或新开 issue，并附上 libpag 版本号、平台与系统版本、可复现的 PAG 文件或最小 demo，我们会优先跟进。（stale-close）
